pkg/shp/cmd/build/upload.go

```diff
@@ -23,16 +24,29 @@ import (
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
+type uploadMethod int
+
+const (
+	streaming uploadMethod = iota
+	bundling
+)
+
 // UploadCommand represents the "build upload" subcommand, implements runner.SubCommand interface.
 type UploadCommand struct {
 	cmd          *cobra.Command              // cobra command instance
 	buildRunSpec *buildv1alpha1.BuildRunSpec // command-line flags stored directly on the BuildRun
-	sourceDir    string                      // local directory to be streamed
 	follow       bool                        // flag to tail pod logs
 
+	buildRefName string // build name
+	sourceDir    string // local directory to be streamed
+
+	method uploadMethod // upload method to be used, either stream or bundle, defaults to stream
```

Comment:
Could we only use `u.sourceBundleImage` to determine the upload method? That would avoid adding extra attributes altogether.

Reply:
I was not sure at first how to "configure" it (so to speak). Probably, I thought it would be easier to read the code when we can use methods in the switch and would enable other options in the future. I will have a look whether we can simplify this one by using basically the field as the decision point.